CRM/Utils/Check/Component.php

```diff
@@ -25,14 +25,44 @@
  +--------------------------------------------------------------------+
  */
 
-
+use Civi\Api4\StatusPreference;
 /**
  *
  * @package CRM
  * @copyright CiviCRM LLC (c) 2004-2019
  */
 abstract class CRM_Utils_Check_Component {
 
+  /**
+   * @var array
+   */
+  public $checksConfig = [];
+
+  /**
+   * Get the configured status checks.
+   *
+   * @return array
+   *
+   * @throws \API_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
+  public function getChecksConfig() {
+    if (empty($this->checksConfig)) {
+      $this->checksConfig = Civi::cache('checks')->get('checksConfig', []);
+      if (empty($this->checksConfig)) {
+        $this->checksConfig = StatusPreference::get()->execute()->indexBy('name');
```

Comment:
@eileenmcnaughton wouldn't you route through civicrm_api4 or similar or something?